### ISSUE TITLE
Vispy error and warning in visulizing 4D time-lapse image and tracks

### ISSUE THREAD

**Issue by pmcesky** [napari/napari#3819]
## 🐛 Bug

I was visualizing my 4D time-lapse image data along with the tracks layer in jupyter-lab. The following error and warning occurred:

![image](https://user-images.githubusercontent.com/18007222/146087074-656764ac-e096-4284-a2dd-525259fef422.png)
![image](https://user-images.githubusercontent.com/18007222/146087097-a6356d9c-c0d9-4ba0-9a23-1058603a39e3.png)

Seems every time when I drag the viewer, as long as the tracks layer is set visible, the error would show up in the terminal and the warning would appear in jupyter-lab. (My tracks layer has a property in array of shape (599,), which is same as the "size=599" in the error message.)

What's wrong? Though I did not find it interferes much with my visualization. How to fix this issue? Or disable it to show the warning messages.

## Environment
napari: 0.4.12
Platform: Windows-10-10.0.19041-SP0
Python: 3.9.7 (default, Sep 16 2021, 16:59:28) [MSC v.1916 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.21.4
SciPy: 1.7.2
Dask: 2021.11.1
VisPy: 0.9.3

OpenGL:
  - GL version:  4.6.13596 Compatibility Profile Context 20.10.20.02 27.20.1020.2002
  - MAX_TEXTURE_SIZE: 16384
 - Any other relevant information:

**Comment by sofroniewn** [comment]
Hmm, I'm not sure about this @JoOkuma has been using the tracks more - maybe he knows? Is there a minimal example - maybe with random data that you can share the reproduces this?

**Comment by jni** [comment]
I haven't seen this using tracks. @pmcesky could you provide some sample data + code that reproduces this?

**Comment by JoOkuma** [comment]
I think there is a mismatch between the tracks data buffer and the [vertex_time buffer](https://github.com/napari/napari/blob/3069f2dcc94a00c9ae4643fd171325ac2afb05dd/napari/_vispy/filters/tracks.py#L152-L155) length.
This is either a bug or it could happen if you're setting the data without using the default API and not updating the `vertex_time` buffer.

**Comment by brisvag** [comment]
This looks very similar to the issue we had in #3623 due to some changes I introduced in #3333. If that's the issue, then I woudl expect it to be broken in 0.4.12 and fixed on `main` by #3729.

**Comment by pmcesky** [comment]
Hi all, 

Thanks very much for the replies!
I found it is probably an issue wrt the graph of the tracking layer. I found it fine without setting the 'graph'. But when 'graph' is added, then this issue occurs. This is mentioned by @quantumjot in https://github.com/napari/napari/issues/3623#issuecomment-983801408.

Some sample data and graph to reproduce this are here:
```
data_sample=np.array([[  1.     ,   0.     , 144.     , 126.     , 109.     ],
       [  1.     ,   1.     , 137.     , 125.     ,  94.     ],
       [  1.     ,   2.     , 128.33333, 124.     ,  90.     ],
       [  2.     ,   0.     , 149.     ,  72.     , 184.     ],
       [  2.     ,   1.     , 149.     ,  60.     , 196.     ]])
graph_sample={2:[1]}
viewer.add_tracks(data=data_sample,graph=graph_sample)
```
The error and warning would occur when loading them into the viewer or dragging the viewer.

**Comment by brisvag** [comment]
Yup, that *definitely* looks like the same issue, so it should already be solved in `main`! :)

**Comment by quantumjot** [comment]
My guess is that this should be fixed by #3729

**Comment by Ciro1990** [comment]
Hi there, I am having the same issue. Has the bug being solved? 

This is what I get: "WARNING: Error drawing visual <vispy.visuals.line.line._GLLineVisual object at 0x000002153284B400>"

**Comment by brisvag** [comment]
@Ciro1990 the bug is already fixed on `main`, but will be available on the next release (which should come in the next couple of weeks!). If you want to get it working now, you can try to checkout the main branch and work from there; otherwise, you need to wait for `0.4.13` to come out :)

**Comment by Ciro1990** [comment]
> @Ciro1990 the bug is already fixed on `main`, but will be available on the next release (which should come in the next couple of weeks!). If you want to get it working now, you can try to checkout the main branch and work from there; otherwise, you need to wait for `0.4.13` to come out :)

Thank you! I will try what you suggested, otherwise I will be waiting for the update :D

**Comment by tlambert03** [comment]
> you can try to checkout the main branch and work from there; otherwise, you need to wait for 0.4.13 to come out :)

Quick reminder that you can always pip install the main branch from GitHub without cloning and checkout:

```
pip install -U git+https://github.com/napari/napari.git
```

**Comment by Ciro1990** [comment]
> pip install -U git+https://github.com/napari/napari.git

Thank you Talley! It worked for me!!!

However, I still have one issue, I can not see my Lineage tree. I uploaded segmentation and tracks, but nothing appears in the lineage tree window. Anybody else has the same issue? 

Could be just me, tho

**Comment by JoOkuma** [comment]
Hi @Ciro1990, are you referring to the track splitting in the napari canvas or the tree lineage from [arboretum](https://github.com/lowe-lab-ucl/arboretum)?

**Comment by Ciro1990** [comment]
> Hi @Ciro1990, are you referring to the track splitting in the napari canvas or the tree lineage from [arboretum](https://github.com/lowe-lab-ucl/arboretum)?

Hi Jo! I meant the lineage tree from arboretum. I just can not manage to make that work

**Comment by JoOkuma** [comment]
I would post an issue there.
It could be that your tracks `graph` is not structured as expected, because this is the only data it uses. Additionally, it doesn't support merges yet.

**Comment by Ciro1990** [comment]
> I would post an issue there. It could be that your tracks `graph` is not structured as expected, because this is the only data it uses. Additionally, it doesn't support merges yet.

Thanks for the inputs, Jo. I will run again the pipeline now. If I still don't get the tree, I will ask for support (biologist struggling with python here :D)

**Comment by andy-sweet** [comment]
Looks like this was fixed by #3729 and maybe again by #4155? @pmcesky , @Ciro1990 : are we good to close this specific issue now?

**Comment by pmcesky** [comment]
> Looks like this was fixed by #3729 and maybe again by #4155? @pmcesky , @Ciro1990 : are we good to close this specific issue now?

Hi Andy, 

I'm not sure about #3729, but the issues I described here and in #4155 were fixed.

**Comment by andy-sweet** [comment]
Thanks for the quick response! In that case, I'm going to close this.

@Ciro1990 : I think your remaining issues are different enough from the original issue described here, but if you feel strongly, please reopen this issue. Or feel free to create a new one!
